pkg/server/grpc_server.go

```diff
@@ -42,6 +42,8 @@ import (
 	"github.com/osrg/gobgp/v3/pkg/packet/bgp"
 )
 
+const defaultListPathBatchSize = 16384
```

Comment:
Why 16k? By default, we should not change the behavior. u64 max?

Reply:
Before 21093fbc8739d0a8e4f84ec6e52c98dad0894612 default was 1 and causes quite slow perfomance. After upgrade gobgp to v3 with n*100K of paths we begin to experience ooms, so it was empirically found 16K by default has still good performance (and allocation one by one is kept for small path ammounts) and have no dramatic memory impact as u64 max which means there'll be no limit by default.
What do you think about to treat here original default as 1 and 16K as feasible (and noticeable) improvement with optional possibility to specify u64 max if one knows what he's doing and has the free memory for such kind of load?

Reply:
The default was 1 three years ago. We set the default to no limit for the last three years. I don't think that we need to change the default behavior.

Reply:
Got it, that you for the suggestion. Please refer change to MaxInt64 by default, it still too high to be used as unlimited and allows to avoid casting len(l) to uint64